test-suite/src/test/scala/org/scalajs/testsuite/compiler/OptimizerTest.scala

```diff
@@ -81,6 +81,64 @@ object OptimizerTest extends JasmineTest {
 
   }
 
+  describe("String_+ constant folding") {
+    it("must fold two constant string to the correct value") {
+      expect("I am " + "constant").toEqual("I am constant")
```

Comment:
Did you check that this is not simply constant-folded by scalac? I think not, but one way to make sure is to extract one of the two sides in an `@inline def`.